### DISCUSSION TITLE
Missing Security Alerts due to large package-lock.json files?

### DISCUSSION THREAD

**Discussion by pixelpogo** [community/community#22143] [New to GitHub]
<p>Hey there,</p>
<p>we have turned on <strong>Security Alerts</strong> for allmost all of our repositories. At some point we saw that some of our <strong>Javascript</strong> repositories show Security Alerts while others do not, although they contain the <strong>exact</strong> same vulnerable version of a library. </p>
<p>It turned out that for some of the repositories the Dependency Graph takes both package.json</p>
<p><strong>and </strong>package-lock.json into account, while for others it is only package.json.</p>
<p>After further investigation we found out that rather big package-lock.json files (e.g.</p>
<p>713KB or  1.01MB in our cases)  seem to be ignored - or they just fail silently.</p>
<p>Our guess is that there is a certain file size threshold or maybe a dependency-complexity threshold for package-lock.json that doesn’t allow Dependency Graph to fully analyze the dependencies. This leads to <strong>missing Security Alerts</strong> for libraries which only show up in the unprocessable package-lock.json files.</p>
<p>Can some-one confirm this behaviour?</p>
<p>If so what is the threshold?</p>
<p>Any ideas to fix or work around this?</p>
<p><span>Thank you very much.</span></p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi there pixelpogo! :wave:</p>
<p>I’ve looked into this for you and I can confirm that there is a size limit on <code>package-lock.json</code> and other manifest files of 0.5 MB, which applies to all users and organizations that are not using GitHub Enterprise.</p>
<p>Manifests that are ignored due to size will not receive security alerts or automated security updates.</p>
<p>We can process up to 20 manifests per repository by default, so if you can split your dependencies between several files that are smaller than 0.5 MB each, they will all be picked up and processed.</p>
<p>One way of doing this would be to use both <code>package-lock.json</code>  and  <code>package.json</code>.</p>
<p>Another way might be to have separate <code>package-lock.json</code> files in sub folders of the repository.</p>
<p>Or, alternately, upgrading to GitHub Enterprise would remove the size limit entirely.</p>
<p>I’ll raise an issue with our docs team to have this limit mentioned on our help pages!</p>
